File: src/SIL.XForge.Scripture/Models/TrainingData.cs
using SIL.XForge.Models;

namespace SIL.XForge.Scripture.Models;

public class TrainingData : ProjectData
{
    public static string GetDocId(string sfProjectId, string dataId) => $"{sfProjectId}:{dataId}";

    public string DataId { get; set; } = string.Empty;
    public string FileUrl { get; set; } = string.Empty;
    public string MimeType { get; set; } = string.Empty;
    public int SkipRows { get; set; }
    public string Title { get; set; } = string.Empty;
}

File: src/RealtimeServer/scriptureforge/models/training-data.ts
import { PROJECT_DATA_INDEX_PATHS, ProjectData } from '../../common/models/project-data';

export const TRAINING_DATA_COLLECTION = 'training_data';
export const TRAINING_DATA_INDEX_PATHS: string[] = PROJECT_DATA_INDEX_PATHS;

export function getTrainingDataId(projectId: string, dataId: string): string {
  return `${projectId}:${dataId}`;
}

export interface TrainingData extends ProjectData {
  dataId: string;
  fileUrl: string;
  mimeType: string;
  skipRows: number;
  title: string;
}
